session-3-backend-api/README.md

```diff
@@ -218,60 +214,52 @@ It formatted the request header and body, and send it to the server using the UR
 ## HTTP Status Code
 Ok, let's say you are uploading a message. How do you know that your message got successfully uploaded?
 
-Then, the __repsonse__ is going to help us!
+The __repsonse__ can tell us!
 
-A HTTP response is just like a request. 
-It has __header__ and __body__.
+A HTTP response is just like a request. It has a __header__ and __body__.
 
 
-In `header`, there is a field called `status` containing a number. That number indicates if your request was successful.
+In the `header`, there is a field called `status` containing a number. That number indicates if your request was successful.
 
 ```
 2XX: The request was successful
 4XX: The request was not successful and client probably messed up.
 5XX: The request was not successful and server probably messed up.
 ```
 
-In `body`, it will conatins whatever data is sent back. 
-It can be JSON/HTML or even random string of text, or nothing at all.
+In the `body`, it will conatin whatever data is sent back. The data can be JSON/HTML, a random string of text, or even nothing at all.
 
 
 Let's check the Chrome developer console and go to `www.google.com`.
 
 * We can find `status` in the `Response Header` tab under the `Header` section.
-* We can also see the `Response` tab to see what is contained in the body
+* We can also see the `Response` tab to see what is contained in the body.
 * The `Preview` tab provides a prettier view of the `body` data.
 
 We can do this in Postman as well.
 
-Let's send a message to `hackschool18.azurewebsites.net/message` but without body.
+Let's send a message to `hackschool18.azurewebsites.net/message` but without a body.
 
 We can see the error code, and nothing is being returned. 
 
 ![400 bad request](images/400postman.png)
 
 ## What is an API?
-API stands for __Application Programming Interface__.
-An API is a definition of methods of communication among various 'components'. 
+API stands for __Application Programming Interface__. An API is a definition of methods of communication among various 'components'. 
 
 
 How is it different from HTTP then? 
-* HTTP defines how computer talk to each other over the world wide web. 
-* A web API defines how frontend code communicates with backend code.
+* HTTP defines how computers can talk to each other over the world wide web. 
+* A web API defines how frontend code can communicate with backend code.
```

Comment:
The difference is one of layering: modern web APIs *build upon* HTTP (and TLS, TCP/IP, etc.) rather than competes against it. Saying it is how frontend communicates with backend isn’t wrong but misses the point, as “frontend communicates with backend through HTTP” is also a true statement.

I’d focus on two things:
* You, the programmer, define the Web APIs you use. On the other hand, HTTP is predefined by engineers of the Internet.
* Web APIs depend on HTTP.